homeassistant/components/tankerkoenig/binary_sensor.py

```diff
@@ -0,0 +1,80 @@
+"""Tankerkoenig binary sensor integration."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ID, ATTR_LATITUDE, ATTR_LONGITUDE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import TankerkoenigDataUpdateCoordinator
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the tankerkoenig sensors."""
+
+    coordinator: TankerkoenigDataUpdateCoordinator = hass.data[DOMAIN][entry.unique_id]
+
+    stations = coordinator.stations.values()
+    entities = []
+    for station in stations:
+        sensor = StationOpenBinarySensorEntity(
+            station,
+            coordinator,
+            coordinator.show_on_map,
```

Comment:
I'm not sure, whether it is a good idea to also have the binary sensors on the map. Isn't the price sensor on the map enough? Otherwise we might be cluttering the map to much

Reply:
TBH since all fuel sensors from same station uses the same coordinates, so they already overlaps each other on the map, so not sure if this frontend "issue" should be solved here? 🤔